src/cli/commands/install.js

```diff
@@ -947,6 +947,17 @@ export async function run(config: Config, reporter: Reporter, flags: Object, arg
   if (flags.lockfile === false) {
     lockfile = new Lockfile();
   } else {
+    const haveLockfile = await fs.exists(path.join(config.lockfileFolder, constants.LOCKFILE_FILENAME));
+    if (haveLockfile) {
+      //no lockfile found, test if node_module exists
+      const haveNodeModules = await fs.exists(path.join(config.lockfileFolder, constants.NODE_MODULES_FOLDER));
+      if (haveNodeModules) {
+        if (!config.nonInteractive && !await reporter.questionAffirm(reporter.lang('lockfileModulesConflict'))) {
+          reporter.footer(false);
```

Comment:
Why are we disabling the footer here?

Also, I think refactoring `questionAffirm` to have a default and handle the promise rejection from `BaseReporter.question` for non-interactive prompts would make this part cleaner.

Reply:
i am addressing this in [PR 5068](https://github.com/yarnpkg/yarn/pull/5068)